### ISSUE TITLE
restriction lint: implicit drop

### ISSUE THREAD

**Issue by lolbinarycat** [rust-lang/rust-clippy#13982]
### What it does

detect all locations where `drop` is called implicitly and suggest turning it into an explicit drop.

possibly has a setting to only warn about locations where a `needs_drop` value is dropped.

### Advantage

would be handy as a learning and debugging tool, allowing you to essentially desugar your code.

### Drawbacks

non-trivial implementation for a lint that will never be used in production code.

### Example

```rust
fn main() {
  let s = "Hello, World".to_string();
  println!("{s}");
}
```

Could be written as:

```rust
fn main() {
  let s = "Hello, World".to_string();
  println!("{s}");
  drop(s);
}
```
